ndmg/scripts/ndmg_dwi_pipeline.py

```diff
@@ -183,107 +184,102 @@ def ndmg_dwi_worker(
+    eddy_corrected_data, bvecs = gen_utils.reorient_dwi(
+        eddy_corrected_data, bvec_scaled, prep_dwi
+    )
 
     # Check dimensions
-    dwi_prep = gen_utils.match_target_vox_res(dwi_prep, vox_size, namer, sens="dwi")
+    eddy_corrected_data = gen_utils.match_target_vox_res(
+        eddy_corrected_data, vox_size, outdir, sens="dwi"
+    )
 
     # Build gradient table
     print("fbval: ", fbval)
     print("bvecs: ", bvecs)
     print("fbvec: ", fbvec)
-    print("dwi_prep: ", dwi_prep)
-    print("namer.dirs: ", namer.dirs["output"]["prep_dwi"])
+    print("eddy_corrected_data: ", eddy_corrected_data)
     gtab, nodif_B0, nodif_B0_mask = gen_utils.make_gtab_and_bmask(
-        fbval, fbvec, dwi_prep, namer.dirs["output"]["prep_dwi"]
+        fbval, fbvec, eddy_corrected_data, str(prep_dwi)
     )
 
     # Get B0 header and affine
-    dwi_prep_img = nib.load(dwi_prep)
-    hdr = dwi_prep_img.header
+    eddy_corrected_data_img = nib.load(str(eddy_corrected_data))
+    hdr = eddy_corrected_data_img.header
 
     # -------- Registration Steps ----------------------------------- #
-    if (skipreg is False) and len(os.listdir(namer.dirs["output"]["prep_anat"])) != 0:
-        try:
-            print("Pre-existing preprocessed t1w files found. Deleting these...")
-            shutil.rmtree(namer.dirs["output"]["prep_anat"])
-            os.mkdir(namer.dirs["output"]["prep_anat"])
-        except:
-            pass
-    if (skipreg is False) and len(os.listdir(namer.dirs["output"]["reg_anat"])) != 0:
-        try:
-            print("Pre-existing registered t1w files found. Deleting these...")
-            shutil.rmtree(namer.dirs["output"]["reg_anat"])
-            os.mkdir(namer.dirs["output"]["reg_anat"])
-        except:
-            pass
-    if (skipreg is False) and (
-        (len(os.listdir(namer.dirs["tmp"]["reg_a"])) != 0)
-        or (len(os.listdir(namer.dirs["tmp"]["reg_m"])) != 0)
-    ):
-        try:
-            print("Pre-existing temporary files found. Deleting these...")
-            shutil.rmtree(namer.dirs["tmp"]["reg_a"])
-            os.mkdir(namer.dirs["tmp"]["reg_a"])
-            shutil.rmtree(namer.dirs["tmp"]["reg_m"])
-            os.mkdir(namer.dirs["tmp"]["reg_m"])
-        except:
-            pass
+
+    # define registration directory locations
+    # TODO: possibly just pull these from a container generated by `gen_utils.make_initial_directories`
+    reg_dirs = ["anat/preproc", "anat/registered", "tmp/reg_a", "tmp/reg_m"]
+    reg_dirs = [outdir / loc for loc in reg_dirs]
+    prep_anat, reg_anat, tmp_rega, tmp_regm = reg_dirs
+
+    if not skipreg:
+        for dir_ in [prep_anat, reg_anat]:
+            if gen_utils.has_files(dir_):
+                gen_utils.as_directory(dir_, remove=True)
+        if gen_utils.has_files(tmp_rega) or gen_utils.has_files(tmp_regm):
+            for tmp in [tmp_regm, tmp_rega]:
+                gen_utils.as_directory(tmp, remove=True)
```

Comment:
I'm 95% sure that this didn't introduce new bugs, but not 100%
(raised 10% from skipeddy because this is literally a direct translation from what it was before, just cleaner, whereas the skipeddy code now does something conceptually slightly different)